### PR TITLE
Added job using default hyperparameters

### PR DESCRIPTION
**Pull request by gauravrajguru**
# PR into Azure/azureml-examples

## Checklist

I have:

- [ ] read and followed the contributing guidelines

## Changes

-

fixes #

### CODE REVIEW

File: sdk/jobs/automl-standalone-jobs/automl-image-object-detection-task-fridge-items/automl-image-object-detection-task-fridge-items-mlflow.ipynb [hunk 1]
```diff
@@ -872,6 +872,79 @@
     "print(resp.text)"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Visualize detections\n",
```
Comment (sharma-riti) [inline sdk/jobs/automl-standalone-jobs/automl-image-object-detection-task-fridge-items/automl-image-object-detection-task-fridge-items-mlflow.ipynb:879]: can we add the visualization section to all tasks same as v1?

File: sdk/jobs/automl-standalone-jobs/automl-image-classification-multiclass-task-fridge-items/automl-image-classification-multiclass-task-fridge-items.ipynb [hunk 2]
```diff
@@ -378,6 +378,82 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# general job parameters\n",
+    "compute_name = \"gpu-cluster\"\n",
```
Comment (sharma-riti) [inline sdk/jobs/automl-standalone-jobs/automl-image-classification-multiclass-task-fridge-items/automl-image-classification-multiclass-task-fridge-items.ipynb:398]: we usually specify the name of the compute in v2 notebooks... I think maybe we should also specify how to create a compute as v1 notebooks? or point to the notebook where compute is generated in the description?

https://github.com/Azure/azureml-examples/blob/main/sdk/resources/compute/compute.ipynb

File: sdk/jobs/automl-standalone-jobs/automl-image-classification-multiclass-task-fridge-items/automl-image-classification-multiclass-task-fridge-items.ipynb [hunk 3]
```diff
@@ -378,6 +378,82 @@
...
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "ml_client.jobs.stream(returned_job.name)"
+   ]
+  },
   {
    "cell_type": "markdown",
    "metadata": {},
```
Comment (PhaniShekhar) [inline sdk/jobs/automl-standalone-jobs/automl-image-classification-multiclass-task-fridge-items/automl-image-classification-multiclass-task-fridge-items.ipynb:459]: I think we can remove the cell that defines compute_name and exp_name in this section since it is already defined above.

File: sdk/jobs/automl-standalone-jobs/automl-image-classification-multiclass-task-fridge-items/automl-image-classification-multiclass-task-fridge-items.ipynb [hunk 4]
```diff
@@ -349,7 +349,7 @@
     "\n",
     "When using AutoML for image tasks, you need to specify the model algorithms using the model_name parameter. You can either specify a single model or choose to sweep over multiple models. Please refer to the [documentation](https://docs.microsoft.com/en-us/azure/machine-learning/how-to-auto-train-image-models?tabs=CLI-v2#configure-model-algorithms-and-hyperparameters) for the list of supported model algorithms.\n",
     "\n",
-    "## 3.1. Configure the job through the image_classification() factory function\n",
+    "## Configure the job through the image_classification() factory function\n",
```
Comment (PhaniShekhar) [inline sdk/jobs/automl-standalone-jobs/automl-image-classification-multiclass-task-fridge-items/automl-image-classification-multiclass-task-fridge-items.ipynb:352]: I am a little confused on where the best place is to put this section. One other alternative is to move this under section 3.1 before the code cells. We can have "image_classification() function parameters:" and "set_limits() parameters:" in that section and have "set_sweep() parameters:" part in the 3.2 section.
@sharma-riti @gauravrajguru  what do you think ?

Comment (PhaniShekhar) [inline sdk/jobs/automl-standalone-jobs/automl-image-classification-multiclass-task-fridge-items/automl-image-classification-multiclass-task-fridge-items.ipynb:352]: Could you please add a small section on set_image_model() parameters: We need not list all the parameters supported. Instead, a small description of it should be good. Maybe we can use the description from this part of the docs: https://docs.microsoft.com/en-us/azure/machine-learning/how-to-auto-train-image-models?tabs=CLI-v2#fixed-settings

File: sdk/jobs/automl-standalone-jobs/automl-image-classification-multiclass-task-fridge-items/automl-image-classification-multiclass-task-fridge-items.ipynb [hunk 5]
```diff
@@ -378,6 +378,82 @@
...
+    "image_classification_job.set_limits(timeout_minutes=60)\n",
+    "\n",
+    "image_classification_job.set_image_model(model_name=\"vitb16r224\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Run the Command\n",
```
Comment (PhaniShekhar) [inline sdk/jobs/automl-standalone-jobs/automl-image-classification-multiclass-task-fridge-items/automl-image-classification-multiclass-task-fridge-items.ipynb:430]: Minor: I think this section from our v1 docs is slightly better. Maybe we can use something like:
Submitting an AutoML job for Computer Vision tasks
Once you've configured your job, you can submit it as a job in the workspace in order to train a vision model using your training dataset.

File: sdk/jobs/automl-standalone-jobs/automl-image-classification-multilabel-task-fridge-items/automl-image-classification-multilabel-task-fridge-items.ipynb [hunk 6]
```diff
@@ -368,6 +368,77 @@
...
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "image_classification_multilabel_job = automl.image_classification_multilabel(\n",
+    "    compute=compute_name,\n",
+    "    experiment_name=exp_name,\n",
+    "    training_data=my_training_data_input,\n",
+    "    validation_data=my_validation_data_input,\n",
+    "    target_column_name=\"label\"\n",
```
Comment (PhaniShekhar) [inline sdk/jobs/automl-standalone-jobs/automl-image-classification-multilabel-task-fridge-items/automl-image-classification-multilabel-task-fridge-items.ipynb:403]: Should we also add primary_metric here ?

File: sdk/jobs/automl-standalone-jobs/automl-image-object-detection-task-fridge-items/automl-image-object-detection-task-fridge-items.ipynb [hunk 7]
```diff
@@ -372,7 +372,7 @@
     "\n",
     "When using AutoML for image tasks, you need to specify the model algorithms using the model_name parameter. You can either specify a single model or choose to sweep over multiple models. Please refer to the [documentation](https://docs.microsoft.com/en-us/azure/machine-learning/how-to-auto-train-image-models?tabs=CLI-v2#configure-model-algorithms-and-hyperparameters) for the list of supported model algorithms.\n",
     "\n",
```
Comment (PhaniShekhar) [inline sdk/jobs/automl-standalone-jobs/automl-image-object-detection-task-fridge-items/automl-image-object-detection-task-fridge-items.ipynb:374]: The v1 notebook also had these sections, could you please add them to this v2 notebook. "Convert annotation file from COCO to JSONL" and "Visualize bounding boxes"

File: sdk/jobs/automl-standalone-jobs/automl-image-object-detection-task-fridge-items/automl-image-object-detection-task-fridge-items-mlflow.ipynb [hunk 8]
```diff
@@ -872,6 +872,79 @@
     "print(resp.text)"
    ]
   },
+  {
```
Comment (PhaniShekhar) [inline sdk/jobs/automl-standalone-jobs/automl-image-object-detection-task-fridge-items/automl-image-object-detection-task-fridge-items-mlflow.ipynb:875]: Minor: Does this visualization work as expected. Just wanted to check since it is something that won't be caught in github actions.

File: sdk/jobs/automl-standalone-jobs/automl-image-object-detection-task-fridge-items/automl-image-object-detection-task-fridge-items-mlflow.ipynb (hunk 1 above)
Comment (PhaniShekhar) [reply sdk/jobs/automl-standalone-jobs/automl-image-object-detection-task-fridge-items/automl-image-object-detection-task-fridge-items-mlflow.ipynb:879]: We need to add model deployment section as well before we can add visualization. 
There is another task to add model deployment and local mlflow inference sections to all tasks. Maybe this can be done as part of that task. What do you think ?

**Review by PhaniShekhar — CHANGES_REQUESTED** [review]

File: sdk/jobs/automl-standalone-jobs/automl-image-object-detection-task-fridge-items/automl-image-object-detection-task-fridge-items-mlflow.ipynb (hunk 1 above)
Comment (gauravrajguru) [reply sdk/jobs/automl-standalone-jobs/automl-image-object-detection-task-fridge-items/automl-image-object-detection-task-fridge-items-mlflow.ipynb:879]: I also think the same. First, we will need to add 2 notebooks for each task i.e., mlflow notebook having inference code and local inference notebook. Then, we can add visualization images to that class. It can be done as a part of that tasks.

Currently, object detection task is having these notebooks. So, I'm adding inferenced image visualization section in them. It can act as reference while adding to other tasks.

File: sdk/jobs/automl-standalone-jobs/automl-image-classification-multilabel-task-fridge-items/automl-image-classification-multilabel-task-fridge-items.ipynb (hunk 6 above)
Comment (gauravrajguru) [reply sdk/jobs/automl-standalone-jobs/automl-image-classification-multilabel-task-fridge-items/automl-image-classification-multilabel-task-fridge-items.ipynb:403]: I think, as we are defining job with default hyperparameters, we can let primary_metric take default value and specify only required values for specified MODEL NAME. 
I checked that primary metric take correct default value for specifica model and task if not given. 

I would like to know your view on this thinking.

File: sdk/jobs/automl-standalone-jobs/automl-image-classification-multiclass-task-fridge-items/automl-image-classification-multiclass-task-fridge-items.ipynb (hunk 2 above)
Comment (gauravrajguru) [reply sdk/jobs/automl-standalone-jobs/automl-image-classification-multiclass-task-fridge-items/automl-image-classification-multiclass-task-fridge-items.ipynb:398]: Added new cell having code to create compute target.

File: sdk/jobs/automl-standalone-jobs/automl-image-object-detection-task-fridge-items/automl-image-object-detection-task-fridge-items-mlflow.ipynb (hunk 8 above)
Comment (gauravrajguru) [reply sdk/jobs/automl-standalone-jobs/automl-image-object-detection-task-fridge-items/automl-image-object-detection-task-fridge-items-mlflow.ipynb:875]: Yes, I checked it on local run.

Comment (gauravrajguru) [reply sdk/jobs/automl-standalone-jobs/automl-image-object-detection-task-fridge-items/automl-image-object-detection-task-fridge-items-mlflow.ipynb:875]: Screenshot 2022-06-29 142840
image

**Review by sharma-riti — APPROVED** [review]

File: sdk/jobs/automl-standalone-jobs/automl-image-object-detection-task-fridge-items/automl-image-object-detection-task-fridge-items.ipynb [hunk 9]
```diff
@@ -311,7 +311,37 @@
...
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Generate jsonl file from coco file\n",
+    "!python coco2jsonl.py \\\n",
+    "--input_coco_file_path \"./odFridgeObjects_coco.json\" \\\n",
+    "--output_dir \"./\" --output_file_name \"odFridgeObjects_from_coco.jsonl\" \\\n",
+    "--task_type \"ObjectDetection\" \\\n",
+    "--base_url uri_folder_data_asset.path"
```
Comment (PhaniShekhar) [inline sdk/jobs/automl-standalone-jobs/automl-image-object-detection-task-fridge-items/automl-image-object-detection-task-fridge-items.ipynb:329]: Shouldn't this be --base_url uri_folder_data_asset.path + "images/" ?
Could you try submitting a run with MLTable referring to the generated jsonl and see if it works.
Similar to the visualization, this is not checked in the GH action. So just to make sure it is working.

Comment (gauravrajguru) [reply sdk/jobs/automl-standalone-jobs/automl-image-object-detection-task-fridge-items/automl-image-object-detection-task-fridge-items.ipynb:329]: Thanks for noticing it Phani. It should be appended with images/. I made changes and also tested with job by making MLtable point to odFridgeObjects_from_coco.jsonl.

Comment (PhaniShekhar) [reply sdk/jobs/automl-standalone-jobs/automl-image-object-detection-task-fridge-items/automl-image-object-detection-task-fridge-items.ipynb]: Is $base_url required ?
